sys/random/fortuna/fortuna.c

```diff
@@ -137,6 +145,26 @@ static int fortuna_pseudo_random_data(fortuna_state_t *state, uint8_t *out,
     return 0;
 }
 
+#if FORTUNA_RESEED_INTERVAL_MS > 0 && IS_USED(MODULE_FORTUNA_RESEED)
+void _reseed_callback(void *arg)
+{
+    fortuna_state_t *state = (fortuna_state_t *) arg;
+    state->needs_reseed = 1;
+}
+
+static void _reseed_timer_init(void) {
+    /* initialize reseed timer */
+    state->reseed_timer.callback = _reseed_callback;
+    state->reseed_timer.arg = state;
+    atomic_store_u8(&state->needs_reseed, 0);
+#if IS_USED(MODULE_ZTIMER_MSEC)
+    ztimer_set(ZTIMER_MSEC, &state->reseed_timer, FORTUNA_RESEED_INTERVAL_MS);
+#else
+    xtimer_set(&state->reseed_timer, FORTUNA_RESEED_INTERVAL_MS * US_PER_MS);
+#endif
```

Comment:
When I mentioned to move this to its own function, I had in mind only this part, so it can be used during initialization and in `fortuna_random_data`

Reply:
Also, state is not defined in this function.